xicam/gui/widgets/plotwidgetmixins.py

```diff
@@ -109,13 +110,114 @@ def showLabel(self, item, points):
         self._curvepoint.setIndex(list(item.scatter.points()).index(point))
 
 
+
+class PlotWidget(QWidget):
+    """Wrapper for pyqtgraph.PlotWidget that allows mixin extensibility.
+
+    Provides three Qt layouts:
+    1. main_layout - vertical box layout that contains the plot view and the bottom widget area
+    2. bottom_layout - vertical box layout that contains extra widgets
+    3. inner_layout - extra horizontal layout inside the bottom layout
+
+    Use the API provided by pyqtgraph.PlotWidget for plotting.
+    """
+    def __init__(self, *args, **kwargs):
+        super(QWidget, self).__init__(*args, **kwargs)
+        self.plot_widget = pg.PlotWidget()
+
+        self.main_layout = QVBoxLayout()
+        self.bottom_layout = QVBoxLayout()
+        self.inner_layout = QHBoxLayout()
+
+        self.main_layout.addWidget(self.plot_widget)
+        self.main_layout.addLayout(self.bottom_layout)
+        self.bottom_layout.addLayout(self.inner_layout)
+
+        self.setLayout(self.main_layout)
+
+    def __getattr__(self, attr):
+        # implicitly wrap methods from plotItem
+        if hasattr(self.plot_widget, attr):
+            m = getattr(self.plot_widget, attr)
+            if hasattr(m, '__call__'):
+                return m
+        raise AttributeError(attr)
+
+
+class AutoRotateCheckbox(PlotWidget):
```

Comment:
maybe we don't need this?

Reply:
this has been removed. after further discussion, the intent of this is not to rotate into a "correct" orientation, but to auto-rotate the viewbox according to screen real estate